bundle/src/main/java/com/meltmedia/dropwizard/etcd/cluster/ClusterAssignmentService.java

```diff
@@ -138,30 +157,29 @@ public ClusterAssignmentService(ScheduledExecutorService executor, ClusterNode t
   public static final String EXCEPTIONS = "exceptions";
   public static final String UNASSIGNMENT_FAILURES = "unassignmentFailures";
   public static final String ASSIGNMENT_FAILURES = "assignmentFailures";
-  public static final String ASSIGNED = "assigned";
-  public static final String TOTAL = "total";
   public static final String CLEAN_UP_TASK = "cleanUpTask";
   public static final String ASSIGNMENT_TASK = "assignmentTask";
-  public static final List<String> ALL_METRICS = Lists.newArrayList(EXCEPTIONS, UNASSIGNMENT_FAILURES, ASSIGNMENT_FAILURES, ASSIGNED, TOTAL, CLEAN_UP_TASK, ASSIGNMENT_TASK);
+  public static final List<String> ALL_METRICS = Lists.newArrayList(EXCEPTIONS, UNASSIGNMENT_FAILURES, ASSIGNMENT_FAILURES, CLEAN_UP_TASK, ASSIGNMENT_TASK);
   
   // extenral dependencies
   ScheduledExecutorService executor;
   ClusterNode thisNode;
   EtcdDirectoryDao<ClusterProcess> processDao;
   ClusterStateTracker stateTracker;
+  Supplier<ProcessorState> processorState;
   MappedEtcdDirectory<ClusterProcess> processDir;
   FixedDelay crashCleanupDelay;
   MetricRegistry registry;
   Function<String, String> metricName;
+  private Supplier<AssignmentState> assignmentState;
 
   // used by the service.
   ScheduledFuture<?> assignmentFuture;
-  WatchService.Watch jobsWatch;
-  AtomicInteger totalProcessCount = new AtomicInteger();
-  AtomicInteger thisProcessCount = new AtomicInteger();
-  Map<String, Set<String>> processes = Maps.newConcurrentMap();
-  List<String> unassigned = Lists.newCopyOnWriteArrayList();
+  volatile long lastAssignmentIndex = 0L;
   private ScheduledFuture<?> cleanupFuture;
+  final Lock emptyLock = new ReentrantLock();
+  final Condition empty  = emptyLock.newCondition();
+  volatile boolean running = false;
```

Comment:
This flag seems like noise.  Watching the state of the active assignments should be enough.